test/update.test.js

```diff
@@ -191,14 +166,17 @@ test("Updates", async (t) => {
 
   await t.test("Routes", async (t) => {
     const res = await fetch(`${address}/`);
+    // Unused - but not unwrapping the response will lead to a dangling promise
+    const _body = await res.text();
```

Comment:
I'm not certain of the issues that may arise from a `dangling promise`, as it doesn't result in any memory leaks.

Reply:
Yeah, it's weird - without that line, `tap` just doesn't exit. With it, it does. Since this is "just" a test I thought it'd be fine to not worry about it too much.

Reply:
Hi @felixrieseberg. I spent two hours investigating this issue, and I have now found the cause.
You have overwritten the local fetch variable in the `src/updates.js` file instead of the global one. This causes the unit tests to use `native fetch` instead of `node-fetch`. Currently, the `native fetch` in Node.js uses `undici`, which is designed for performance and requires consuming the response body (i.e., `res.text()` / `res.json()`). This is not a bug but rather a feature of `undici`.

Based on these points, we have the following solutions:

1. Change `let fetch = global.fetch;` to `fetch = global.fetch;`
2. Add `fetch = require("node-fetch").default;` before ``const res = await fetch(`${address}/`, {``
3. Use `resp.text()` (no need for await or assigning to _body variable)
4. Remove the contents of https://github.com/electron/update.electronjs.org/pull/133/files#diff-7e34426767a744becfe2d43d1b99a5b1c5d1373606e69be0e723719729c480d2R14-R22 and add `const fetch = require("node-fetch").default` to each unit test.


Ref: https://github.com/nodejs/undici/issues/1203#issuecomment-1398207382

Reply:
Thank you for your research! I went with option (3).